src/bvals/comms/bvals_utils.hpp

```diff
@@ -103,6 +103,9 @@ void InitializeBufferCache(std::shared_ptr<MeshData<Real>> &md, COMM_MAP *comm_m
   pcache->buf_vec.clear();
   pcache->idx_vec = std::vector<std::size_t>(key_order.size());
   std::for_each(std::begin(key_order), std::end(key_order), [&](auto &t) {
+    if (comm_map->count(std::get<2>(t)) == 0) {
+      PARTHENON_FAIL("Asking for buffer that doesn't exist");
+    }
```

Comment:
What's going on here?

Reply:
This was part of a debugging "exercise" and I saw no reason to remove it once things were fixed. It is just checking that a comm buffer corresponding to the requested key exists. I think previously the map would throw an error if you asked for a non-existent key, but now we can't trust exceptions.

Reply:
:+1: